From: Jeff Mahoney <jeffm@suse.com>
Subject: ext4: force read-only unless rw=1 module option is used
References: fate#311111 fate#314864
Patch-mainline: never

SLE11 only supports ext4 in read-only mode for migration purposes.
This patch forces mounts read-only and issues a message explaining why.

However, since some users still need to use ext4 in read-write mode,
we'll allow read-write mounts when the rw=1 module parameter is passed
during module load. This will taint the kernel and module as unsupported
and also issues a message explaining why.

This can eliminate the ext4-writable KMP implementation by just
requiring the 'options ext4 rw=1' line in a file in /etc/modprobe.d.

However, if there are both SLE 11 SP2 and SP3 kernels installed, there
may arise a conflict so we can't just convert ext4-writeable to add that
line and replace the KMP. Thus, if ext4 is built using as the
ext4-writeable KMP, it will set allow_rw to true by default.

Signed-off-by: Jeff Mahoney <jeffm@suse.com>
---
 fs/ext4/super.c |   59 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 59 insertions(+)

--- a/fs/ext4/super.c
+++ b/fs/ext4/super.c
@@ -5031,3 +5043,7 @@ MODULE_DESCRIPTION("Fourth Extended File
 MODULE_LICENSE("GPL");
 module_init(ext4_init_fs)
 module_exit(ext4_exit_fs)
+
+static bool rw;
+module_param(rw, bool, 0644);
+MODULE_PARM_DESC(rw, "Allow read-write file systems (ignored; for compatibility with SLE11)");
